### DISCUSSION TITLE
Add text / annotation to tar_visnetwork()

### DISCUSSION THREAD

**Discussion by mbrousil** [ropensci/targets#603] [Help] [answered]
I'd like to add a timestamp to the output from my `tar_visnetwork()` call. Is there a way to do this? I had in mind something that performs like `ggtitle()` or `ggplot2::annotate()`.

Thanks!

**Comment by wlandau** [comment]
`tar_visnetwork()` returns a `visNewtork` object, and you are free to modify it. (See the reprex below.) It is also straightforward (and fun) to create your own `visNetwork` graph using the nodes and edges generated by `tar_network()`. I hesitate to add native flexibility to `targets` because it adds a maintenance cost for something that is not a common use case.

``` r
library(targets)
tar_script()
graph <- tar_visnetwork()
graph$x$nodes$label <- paste(graph$x$nodes$label, "\ncustom")
graph
```

![](https://i.imgur.com/EzkTyU0.png)

<sup>Created on 2021-08-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Reply by wlandau** [reply]
For completeness: `tar_meta()$time` gives you timestamps, and you can `dplyr::left_join()` those to the nodes data frame.

**Reply by mbrousil** [reply]
Ah, thank you! I wasn't familiar with `visNetwork` so I will work with that. Totally understand the need to avoid additional maintenance cost.

What I had in mind was adding a single label or title to the top of the plot as opposed to labels for each node. However, the node-by-node approach could work as well. Looking at the structure of `visNetwork` objects, I can get what I want like this:

``` r
library(targets)

targets::tar_script(ask = FALSE)

graph <- tar_visnetwork()

graph$x$main$text <- paste0("Last built: ", Sys.time())

graph

```
![Rplot](https://user-images.githubusercontent.com/37380883/129627384-bb9f7d34-3f36-4b6e-ac10-a9480f9a8e05.png)

<sup>Created on 2021-08-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
 
Not sure if this is the cleanest way to go about it, though.

**Reply by wlandau** [reply]
Seems fine if you are just making these graphs for yourself (as opposed to a reusable package you will publish on CRAN). I did not see a function in `visNetwork` that would allow you to add a title to an already constructed graph object. `visNetwork::visSetTitle()` seems to only work for their built-in Shiny app.

**Reply by mbrousil** [reply]
Yep, I had held out hope for that function as well. Thanks for your help!

**Comment by mbrousil** [comment, marked as answer]
Issue solved with `graph$x$main$text <- paste0("Last built: ", Sys.time())`
